src/lib.rs

```diff
@@ -174,7 +182,9 @@
        html_favicon_url = "https://www.rust-lang.org/favicon.ico",
        html_root_url = "https://docs.rs/rand/0.5")]
 
+#![deny(missing_docs)]
 #![deny(missing_debug_implementations)]
+#![doc(test(attr(allow(unused_variables), deny(warnings))))]
```

Comment:
I generally prefer warnings to errors (`#![warn(...)]`) simply because new compiler versions may add warnings without considering backwards compatibility, thus breaking builds. In the latter case unfortunately we have no choice since warnings from examples are not shown; also "missing docs" is not something the compiler can break so that's fine.

Reply:
Good point, I'll keep that in mind!